[Lin Zechen] iP

Bubbl3T opened the pull request:
# DukeTasks

* This are bullet points created with "*"
* They are similar to Markdown.

> More information on GFMD [here](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax). 😉 

Numbered lists are like so:

1. First
1. Second
1. Third

```java
// This is a fenced code block with comments.
public class Main {
    public static void main(String[] args) {
        System.out.println("Hello World!");
    }
}
```

Todo:

- [x] individual project
- [ ] team project

Emoji 🥇 

Inline `code` is easy!

Text formatting: **bold,** _italics_, ~~strike-through~~

Commit:
Add Gradle support

Bubbl3T committed:
Added basic skeleton to Duke and enabled user input.

Bubbl3T committed:
Added feature to store items in a list and print the entire list.

Bubbl3T committed:
Added feature to mark items in the list as done and not done.

Bubbl3T committed:
Abstracted Task as a class.

Bubbl3T committed:
Added variations to the type of tasks that can be added.

Bubbl3T committed:
Created test case for TextUiTesting

Bubbl3T committed:
Added exception handling.

Bubbl3T committed:
Added delete feature and changed the storage for task from array to ArrayList

Bubbl3T committed:
Add Enum in implementation

Bubbl3T committed:
Add the save feature to Duke

Bubbl3T committed:
Update gitignore

Bubbl3T committed:
Update save feature to save whenever the list changes

The save feature was only able to save when the bye command is
called.

This can cause data to be lost if program is terminated
by other means or when errors occur.

Let's store the data whenever the list changes to prevent data loss
due to unexpected errors.

Bubbl3T committed:
Update gitignore

Bubbl3T committed:
Change Deadline and Event to store datetime as LocalDateTime object

Bubbl3T committed:
Merge branch 'branch-Level-7'

Bubbl3T committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

Bubbl3T committed:
Fix bug that occurred from merging commits

Bubbl3T committed:
Update OOP and file structure

OOP concepts were not strongly applied, causing code to be all
around the place. Files were all under the same directory without
proper organization.

This made editing and debugging tedious and inefficient.

Let's apply the OOP concepts to properly organize the code, and
place the files under different packages to improve organization.

Bubbl3T committed:
Fix unintended changes made to text files from refactoring

Bubbl3T committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip into branch-add-gradle-support

Bubbl3T committed:
Merge branch 'branch-add-gradle-support'

Bubbl3T committed:
Add a few test cases and bugfix

Bubbl3T committed:
Update gradle settings to create jar file

Bubbl3T committed:
Add find command to Duke

Bubbl3T committed:
Change variable and method names to match coding standards

Bubbl3T committed:
Change wildcard import to list imports explicitly

Bubbl3T committed:
Add Javadoc comments to Duke and DukeException classes

Bubbl3T committed:
Add Javadoc comments to most public classes.

Bubbl3T committed:
Merge branch 'branch-Level-9'

Bubbl3T committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/commands/ListCommand.java
#	src/main/java/duke/tools/Ui.java

Bubbl3T committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/commands/ByeCommand.java

Bubbl3T committed:
Add more Javadoc comments

Bubbl3T committed:
Add more Javadoc and fix mistakes in naming convention

Bubbl3T committed:
Apply Checkstyle and fix style violations

Bubbl3T committed:
Add Gui class and test JavaFX working condition

Bubbl3T committed:
Add basic interface to Duke

Bubbl3T committed:
Create Gui for Duke

Bubbl3T committed:
Fix styling issues

Bubbl3T committed:
Apply FXML to GUI

Bubbl3T committed:
Merge branch 'branch-Level-10'

gnahzeus commented on src/main/java/duke/tools/Parser.java:
```diff
@@ -0,0 +1,142 @@
+     *
+     * @param str The input String from the user specifying the parameters of the event.
+     * @returnThe EventCommand representing the user's input
+     * @throws DukeException
```
Should a javadocs description be included for the parameter `DukeException`?

gnahzeus commented on src/main/java/duke/tools/Parser.java:
```diff
@@ -0,0 +1,142 @@
+     * Makes sense of the user's input to add an event.
+     *
+     * @param str The input String from the user specifying the parameters of the event.
+     * @returnThe EventCommand representing the user's input
```
Might there be an intention to put a space between the param and the description?

Bubbl3T replied:
Thanks for pointing out my styling errors in such detail! I would definitely have missed that myself. 😄

gnahzeus commented on src/main/java/duke/tools/Parser.java:
```diff
@@ -0,0 +1,142 @@
+public class Parser {
+
+    /** The user's input date and time format. */
+    private static final DateTimeFormatter DT_PARSE_FORMAT = DateTimeFormatter.ofPattern("[yyyy][yy]-M-d HHmm");
+    /** The date and time format printed out by Duke. */
+    private static final DateTimeFormatter DT_PRINT_FORMAT = DateTimeFormatter.ofPattern("MMM dd yyyy, KK:mm a");
```
I like that your associated constants have a common prefix :+1:

gnahzeus commented on src/main/java/duke/tools/Storage.java:
```diff
@@ -0,0 +1,194 @@
+                    + "/by " + deadline.getDateTime() + System.lineSeparator();
+        case EVENT:
+            Event event = (Event) task;
+            return "E " + (event.isDone() ? "Y " : "N ") + event.getDescription()
+                    + "/at " + event.getDateTime() + System.lineSeparator();
```
I like how you used line wrapping at appropriate places to keep the line length short!

gnahzeus reviewed (commented):
Hey @Bubbl3T, overall, I found your code easy to read, and your variables and methods soundly named. Besides a few nits with the javadocs, this looks good!  :+1:

Bubbl3T committed:
Reduce number of files needed to manage GUI

jasonchristopher21 commented on src/main/java/duke/tools/SessionManager.java:
```diff
@@ -0,0 +1,51 @@
+    private SessionManager() {}
+
+    /**
+     * Initialize the SessionManager with the TaskList and Storage for the session.
```
```suggestion
     * Initializes the SessionManager with the TaskList and Storage for the session.
```
Perhaps Intialize can better be worded as Initializes?

jasonchristopher21 reviewed (commented):
Your code looks really good 😺 👍 
Minor spelling change requested

Bubbl3T committed:
Add assertions to Session Manager

There were no assertion statements in the program.

Assertions help to check if our program is running with our
intended assumptions.

Let's add some assertion statements so our program runs smoothly
and become easier to bug fix in the future.

Bubbl3T committed:
Ensure code quality is maintained

Code quality mostly maintained, but can be done slightly better.

Good code quality helps to make the code more readable and easier
to understand.

Let's fix common errors in code quality to make code more readable.

Bubbl3T committed:
Use streams when dealing with the list of tasks

For loop was used in iterating through the list of tasks.

While for loop is usable, it might run into issues with index out
of bounds when we try to retrieve items from a list using a for
loops.

Let's use streams to manage lists of tasks instead.

Doing so helps us reduce human error and makes code easier to
understand as well.

Bubbl3T committed:
Merge pull request #1 from Bubbl3T/branch-A-Assertions

Add assertions to Session Manager

Bubbl3T committed:
Merge pull request #2 from Bubbl3T/branch-A-Streams

Use streams when dealing with the list of tasks

Bubbl3T committed:
Reduce the use of magic strings

There are some strings in the program that are added without any
information of what those strings represent.

This can make changing those strings in the future susceptible to
bugs.

Let's make those strings final and give them variable names instead
to improve code quality.

Bubbl3T committed:
Merge pull request #3 from Bubbl3T/branch-A-CodeQuality

Improve Code Quality

Bubbl3T committed:
Fix error in code styling

Bubbl3T committed:
Fix error in commit

Bubbl3T committed:
Create gradle.yml

Bubbl3T committed:
Remove unintended file

Bubbl3T committed:
Merge remote-tracking branch 'origin/master'

Bubbl3T committed:
Add new feature to search for tasks between 2 date times.

Bubbl3T committed:
Add feature to reschedule task

Bubbl3T committed:
Create more exception classes

Bubbl3T committed:
Improve GUI to look better

Bubbl3T committed:
Fix Gui bug and allow window resizing

Bubbl3T committed:
Create the Ui screenshot

Bubbl3T committed:
Add new test cases and improve Gui

Some test cases were outdated due to the change in exception
handling. Also, the Gui previously show the entire text box around
the picture too, making the screen look cluttered.

Let's add some new test cases and fix the old test cases to better
test our code. Let's also change the Gui so that the textbox only
appear beside the picture in a smaller size.

Bubbl3T committed:
Add more test cases

Bubbl3T committed:
Add User Guide for Duke

Bubbl3T committed:
Add improvements to how Shiro speaks

Bubbl3T committed:
Fix minor bugs before release

Bubbl3T committed:
Rename image files

Bubbl3T committed:
Update gitignore

Bubbl3T committed:
Update gradle.yml

Bubbl3T committed:
Update User Guide

Bubbl3T committed:
Fix bugs regarding images not displaying correctly

Bubbl3T committed:
Fix User Guide bug where image will not be display

Bubbl3T committed:
Update github workflow

Bubbl3T committed:
Update gradlew to fix CI on github

Bubbl3T committed:
Update gradlew to have modified access

Bubbl3T committed:
Add checkstyleMain and checkstyleTest

Bubbl3T committed:
Fix bug of CI not running correctly
